### PR TITLE
add v container util

### PR DESCRIPTION
**Pull request by phuongdmp7**

### CODE REVIEW

File: Scripts/Utilities/Extension/VContainerUtils.cs
Comment (frostbun) [inline Scripts/Utilities/Extension/VContainerUtils.cs:1]: move to Scripts/DI/VContainer/VContainerWrapper.cs

Comment (phuongdmp7) [reply Scripts/Utilities/Extension/VContainerUtils.cs:1]: That path can't use GetDerivedTypes. If reference GameFoundation.Utilities will be a circular dependency

File: Scripts/Utilities/Extension/VContainerUtils.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#if GDK_VCONTAINER
+namespace GameFoundation.Scripts.Utilities.Extension
+{
+    using System.Linq;
+    using VContainer;
+
+    public static class VContainerUtils
+    {
+        /// <summary>
+        /// Same as Zenject's ReBind method.
```
Comment (frostbun) [inline Scripts/Utilities/Extension/VContainerUtils.cs]: this is not the same as ReBind, this a replacement

File: Scripts/Utilities/Extension/VContainerUtils.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        /// </summary>
+        public static void RegisterWithDerivedTypes<T>(this IContainerBuilder builder, Lifetime lifetime = Lifetime.Singleton)
```
Comment (frostbun) [inline Scripts/Utilities/Extension/VContainerUtils.cs]: RegisterFromDerivedType

**Review by frostbun — APPROVED** [review]

**Pull request merged**